overlore/sqlite/townhall_db.py

```diff
@@ -0,0 +1,186 @@
+    #     if limit <= 0:
+    #         raise ValueError("Limit must be higher than 0")
+    #     if self.get_entries_count() == (0, 0):
+    #         return []
+
+    #     query = """
+    #         SELECT v.rowid, v.distance FROM vss_npc_thought v
+    #         INNER JOIN npc_thought t ON v.rowid = t.rowid
+    #         WHERE t.npc_entity_id = ? AND vss_search(embedding, vss_search_params(?, ?))
+    #     """
+
+    #     values = (npc_entity_id, json.dumps(query_embedding), limit + 1)
+
+    #     return self.execute_query(query, values)
+
+    def query_cosine_similarity(self, query_embedding: list[float], npc_entity_id: int, limit: int = 1) -> StoredVector:
+        if limit <= 0:
+            raise ValueError("Limit must be higher than 0")
+
+        query = """
+            SELECT vss.rowid, vss_cosine_similarity(?, embedding) AS similarity
+            FROM vss_npc_thought vss
+            INNER JOIN npc_thought t ON vss.rowid = t.rowid
+            WHERE t.npc_entity_id = ?
+            ORDER BY similarity DESC
+            LIMIT ?;
+        """
+        values = (json.dumps(query_embedding), npc_entity_id, limit)
+        res = self.execute_query(query, values)
+
+        if not res:
+            raise CosineSimilarityNotFoundError(f"No similar thought found for npc entity {npc_entity_id}")
+        return cast(StoredVector, res[0])
+
+    def insert_townhall_discussion(self, realm_id: int, discussion: str, townhall_input: str) -> int:
+        discussion = discussion.strip()
+        ts = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        return self._insert(
+            "INSERT INTO townhall (discussion, townhall_input, realm_id, ts) VALUES (?, ?, ?, ?);",
+            (discussion, townhall_input, realm_id, ts),
+        )
+
+    def fetch_realm_townhalls(self, realm_id: int) -> list:
+        return self.execute_query(
+            "SELECT discussion, townhall_input FROM townhall WHERE realm_id = ? ORDER BY ts DESC;",
+            (realm_id,),
+        )
+
+    def insert_plotline(self, realm_id: int, plotline: str) -> int:
+        return self._insert("INSERT INTO realm_plotline (plotline, realm_id) VALUES (?, ?);", (plotline, realm_id))
+
+    def fetch_plotline_by_realm(self, realm_id: int) -> str:
+        query = """
+            SELECT plotline FROM realm_plotline
+            WHERE realm_id = ?;
+        """
+        values = (realm_id,)
+        res = self.execute_query(query, values)
+        if res == []:
+            return ""
+        return cast(str, res[0][0])
+
+    def update_plotline(self, realm_id: int, new_plotline: str) -> int:
+        return self._update("UPDATE realm_plotline SET plotline = ? WHERE realm_id = ?;", (new_plotline, realm_id))
+
+    def delete_plotline(self, realm_id: int) -> None:
+        self.execute_query("DELETE FROM realm_plotline WHERE realm_id = ?", (realm_id,))
```

Comment:
Need to commit your change here, why not use the _delete function?

Reply:
Good find, changed it